### ISSUE TITLE
upgrade to v 0.18.0.0 fails

### ISSUE THREAD

**Issue by dawiepoolman** [monero-ecosystem/PiNode-XMR#80]
My node is on v5.22.07-Open-Build
I say this in the cmd line:

mv: cannot stat '/home/pinodexmr/temp/monero-arm-linux-gnueabihf-v0.17.3.2/monero*': No such file or directory
mkdir: cannot create directory ‘.bitmonero’: File exists

I have rebuild from source as well but node wont start.  Yhe node status shows:

Tor Public  :  activating (auto-restart) (Result: exit-code) since Thu 2022-07-21 21:35:36 SAST; 26s ago

log.io that could help:

2022-07-21 19:41:18.566	        b6f58ba0	ERROR	msgwriter	src/common/scoped_message_writer.h:102	Error: Couldn't connect to daemon: 192.168.1.20:18081
2022-07-21 19:41:31.152	        b6edeba0	INFO	logging	contrib/epee/src/mlog.cpp:273	New log categories: *:WARNING,net:FATAL,net.http:FATAL,net.ssl:FATAL,net.p2p:FATAL,net.cn:FATAL,daemon.rpc:FATAL,global:INFO,verify:FATAL,serialization:FATAL,daemon.rpc.payment:ERROR,stacktrace:INFO,logging:INFO,msgwriter:INFO
2022-07-21 19:41:31.153	        b6edeba0	INFO	logging	contrib/epee/src/mlog.cpp:273	New log categories: *:INFO,global:INFO,stacktrace:INFO,logging:INFO,msgwriter:INFO,perf.*:DEBUG
2022-07-21 19:41:31.153	        b6edeba0	INFO	global	src/daemon/main.cpp:296	Monero 'Fluorine Fermi' (v0.18.0.0-release)

**Comment by dawiepoolman** [comment]
extra two log lines show another warning that I might have stuffed up w.r.t bind address?:

2022-07-21 19:41:31.153	        b6edeba0	INFO	global	src/daemon/main.cpp:296	Monero 'Fluorine Fermi' (v0.18.0.0-release)
2022-07-21 19:41:31.154	        b6edeba0	INFO	daemon	src/daemon/main.cpp:358	Moving from main() into the daemonize now.
2022-07-21 19:41:31.154	        b6edeba0	WARNING	daemon	src/daemon/main.cpp:103	--public-node is enabled, but RPC server 192.168.1.20:18081 may be unreachable from outside, please check RPC server bind address

**Comment by dawiepoolman** [comment]
something very strange in my logs.  It runs for a few seconds and end in a loop with this


2022-07-23 01:45:28.283	        b6fa7ba0	INFO	stacktrace	src/common/stack_trace.cpp:134	Unwound call stack:
2022-07-23 01:45:28.284	        b6fa7ba0	INFO	stacktrace	src/common/stack_trace.cpp:172	    [1]  0xab) [0x4507bc]:__cxa_throw+0xab) [0x4507bc]
2022-07-23 01:45:28.284	        b6fa7ba0	INFO	stacktrace	src/common/stack_trace.cpp:172	    [2] ./monero/build/release/bin/monerod(+0x33afa) [0x453afa] 
2022-07-23 01:45:28.285	        b6fa7ba0	INFO	stacktrace	src/common/stack_trace.cpp:172	    [3] ./monero/build/release/bin/monerod(+0x47f45a) [0x89f45a] 
2022-07-23 01:45:28.285	        b6fa7ba0	INFO	stacktrace	src/common/stack_trace.cpp:172	    [4] ./monero/build/release/bin/monerod(+0x47f7b6) [0x89f7b6] 
2022-07-23 01:45:28.285	        b6fa7ba0	INFO	stacktrace	src/common/stack_trace.cpp:172	    [5] ./monero/build/release/bin/monerod(+0x6b5d8) [0x48b5d8] 
2022-07-23 01:45:28.285	        b6fa7ba0	INFO	stacktrace	src/common/stack_trace.cpp:172	    [6] ./monero/build/release/bin/monerod(+0x9d900) [0x4bd900] 
2022-07-23 01:45:28.285	        b6fa7ba0	INFO	stacktrace	src/common/stack_trace.cpp:172	    [7] ./monero/build/release/bin/monerod(+0xc6d12) [0x4e6d12] 
2022-07-23 01:45:28.285	        b6fa7ba0	INFO	stacktrace	src/common/stack_trace.cpp:172	    [8] ./monero/build/release/bin/monerod(+0xc7458) [0x4e7458] 
2022-07-23 01:45:28.285	        b6fa7ba0	INFO	stacktrace	src/common/stack_trace.cpp:172	    [9] ./monero/build/release/bin/monerod(+0xf6e26) [0x516e26] 
2022-07-23 01:45:28.285	        b6fa7ba0	INFO	stacktrace	src/common/stack_trace.cpp:172	    [10] ./monero/build/release/bin/monerod(+0xaba42) [0x4cba42] 
2022-07-23 01:45:28.285	        b6fa7ba0	INFO	stacktrace	src/common/stack_trace.cpp:172	    [11] ./monero/build/release/bin/monerod(+0x54f9a) [0x474f9a] 
2022-07-23 01:45:28.285	        b6fa7ba0	INFO	stacktrace	src/common/stack_trace.cpp:172	    [12] ./monero/build/release/bin/monerod(+0x51698) [0x471698] 
2022-07-23 01:45:28.285	        b6fa7ba0	INFO	stacktrace	src/common/stack_trace.cpp:172	    [13] ./monero/build/release/bin/monerod(+0x36676) [0x456676] 
2022-07-23 01:45:28.285	        b6fa7ba0	INFO	stacktrace	src/common/stack_trace.cpp:172	    [14] /lib/arm-linux-gnueabihf/libc.so.6(+0x1d7d6) [0xb65cd7d6] 
2022-07-23 01:45:28.285	        b6fa7ba0	INFO	stacktrace	src/common/stack_trace.cpp:172	    [15]  0x5d) [0xb65cd886]:__libc_start_main+0x5d) [0xb65cd886]
2022-07-23 01:45:28.285	        b6fa7ba0	INFO	stacktrace	src/common/stack_trace.cpp:172	
2022-07-23 01:45:28.286	        b6fa7ba0	INFO	stacktrace	src/common/stack_trace.cpp:133	Exception: boost::exception_detail::clone_impl<boost::system::system_error>
2022-07-23 01:45:28.286	        b6fa7ba0	INFO	stacktrace	src/common/stack_trace.cpp:134	Unwound call stack:
2022-07-23 01:45:28.286	        b6fa7ba0	INFO	stacktrace	src/common/stack_trace.cpp:172	    [1]  0xab) [0x4507bc]:__cxa_throw+0xab) [0x4507bc]
2022-07-23 01:45:28.286	        b6fa7ba0	INFO	stacktrace	src/common/stack_trace.cpp:172	    [2] ./monero/build/release/bin/monerod(+0x33a02) [0x453a02] 
2022-07-23 01:45:28.286	        b6fa7ba0	INFO	stacktrace	src/common/stack_trace.cpp:172	    [3] ./monero/build/release/bin/monerod(+0xbea8c) [0x4dea8c] 
2022-07-23 01:45:28.286	        b6fa7ba0	INFO	stacktrace	src/common/stack_trace.cpp:172	    [4] ./monero/build/release/bin/monerod(+0xc3944) [0x4e3944] 
2022-07-23 01:45:28.287	        b6fa7ba0	INFO	stacktrace	src/common/stack_trace.cpp:172	    [5] ./monero/build/release/bin/monerod(+0xc6d94) [0x4e6d94] 
2022-07-23 01:45:28.287	        b6fa7ba0	INFO	stacktrace	src/common/stack_trace.cpp:172	    [6] ./monero/build/release/bin/monerod(+0xc7458) [0x4e7458] 
2022-07-23 01:45:28.287	        b6fa7ba0	INFO	stacktrace	src/common/stack_trace.cpp:172	    [7] ./monero/build/release/bin/monerod(+0xf6e26) [0x516e26] 
2022-07-23 01:45:28.287	        b6fa7ba0	INFO	stacktrace	src/common/stack_trace.cpp:172	    [8] ./monero/build/release/bin/monerod(+0xaba42) [0x4cba42] 
2022-07-23 01:45:28.287	        b6fa7ba0	INFO	stacktrace	src/common/stack_trace.cpp:172	    [9] ./monero/build/release/bin/monerod(+0x54f9a) [0x474f9a] 
2022-07-23 01:45:28.287	        b6fa7ba0	INFO	stacktrace	src/common/stack_trace.cpp:172	    [10] ./monero/build/release/bin/monerod(+0x51698) [0x471698] 
2022-07-23 01:45:28.287	        b6fa7ba0	INFO	stacktrace	src/common/stack_trace.cpp:172	    [11] ./monero/build/release/bin/monerod(+0x36676) [0x456676] 
2022-07-23 01:45:28.287	        b6fa7ba0	INFO	stacktrace	src/common/stack_trace.cpp:172	    [12] /lib/arm-linux-gnueabihf/libc.so.6(+0x1d7d6) [0xb65cd7d6] 
2022-07-23 01:45:28.287	        b6fa7ba0	INFO	stacktrace	src/common/stack_trace.cpp:172	    [13]  0x5d) [0xb65cd886]:__libc_start_main+0x5d) [0xb65cd886]
2022-07-23 01:45:28.287	        b6fa7ba0	INFO	stacktrace	src/common/stack_trace.cpp:172	
2022-07-23 01:45:28.288	        b6fa7ba0	ERROR	msgwriter	src/common/scoped_message_writer.h:102	Error: Couldn't connect to daemon: 192.168.1.20:18081
2022-07-23 01:45:48.390	        b6ecbba0	INFO	logging	contrib/epee/src/mlog.cpp:273	New log categories: *:WARNING,net:FATAL,net.http:FATAL,net.ssl:FATAL,net.p2p:FATAL,net.cn:FATAL,daemon.rpc:FATAL,global:INFO,verify:FATAL,serialization:FATAL,daemon.rpc.payment:ERROR,stacktrace:INFO,logging:INFO,msgwriter:INFO

**Comment by dawiepoolman** [comment]
odd observation.. there are multiple processids spawned form monerod (like 18 originally) for user pinodexmr.  I saw user root also had a monerod PID.
Every few seconds the service restarts with a new id.  Is there some hidden systemd level auto reboot on failure service? 
I also saw that within the .bitcoind/lmdb dir the permission was set for root.  I chown it to pinodexmr instead.

I stil se on the Nodestatus dasboad the line:
Private Node:  activating (auto-restart) (Result: signal) since Sat 2022-07-23 11:04:58 SAST; 3s ago

why so unstable?

**Comment by dawiepoolman** [comment]
I think the monero build was corrupted tbh.  When rebuilding with a forced upgrade the process is riddled with errors and finished way too fast (like 15min) on a RPi3

**Comment by dawiepoolman** [comment]
this hard coded line seems wrong?
![image](https://user-images.githubusercontent.com/2351212/180661632-7a475100-1605-4089-a627-70f788164684.png)

**Comment by dawiepoolman** [comment]
ok, I manualy downloaded and replaced the arm7 binaries for v0.18.0.0 and it seems to be running fine now.
I configured file moneroCustomNode.sh just to be save as well to run monerod 'manually'.

**Comment by shermand100** [comment]
Sorry for not getting to you sooner, I've been away for the weekend.

So as you gathered the Monero 'release-v0.18' was released but it takes the dev team a few days to put together the binaries that you use.

Frustratingly  I had made those changes 4 days ago (https://github.com/monero-ecosystem/PiNode-XMR/commit/bef3c44c2368485a04fcff89b0e924e0368616c8) but require a user to do an 'update PiNodeXMR' before 'Update Monero' to catch that change.
Sorry for the inconvenience.

Going forward I'll move all updater scripts online so reactive changes such as this don't require a PiNodeXMR to take effect.
